src/command_modules/azure-cli-natgateway/azure/cli/command_modules/natgateway/custom.py

```diff
@@ -0,0 +1,59 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from azure.cli.core.util import sdk_no_wait
+
+from knack.log import get_logger
+from knack.util import CLIError
+
+logger = get_logger(__name__)
+
+
+def network_client_factory(cli_ctx, **kwargs):
+    from azure.cli.core.profiles import ResourceType
+    from azure.cli.core.commands.client_factory import get_mgmt_service_client
+    return get_mgmt_service_client(cli_ctx, ResourceType.MGMT_NETWORK, **kwargs)
+
+
+def create_nat_gateway(cmd, nat_gateway_name, resource_group_name,
+                       location=None, public_ip_addresses=None,
+                       public_ip_prefixes=None, idle_timeout=None, no_wait=False):
+
+    if not public_ip_addresses and not public_ip_prefixes:
+        raise CLIError('usage error: --public-ip-addresses ADDRESSES | --public-ip-prefixes PREFIXES')
+
+    if public_ip_addresses is not None and public_ip_prefixes is not None:
+        logger.error("public_ip_addresses OR public_ip_prefixes required")
```

Comment:
Why not raise error? If both public ip and public ip prefixes passed in, which one will take high priority?